powerauth-restful-security-base/src/main/java/io/getlime/security/powerauth/rest/api/base/provider/PowerAuthEncryptionProviderBase.java

```diff
@@ -46,36 +59,79 @@
     public abstract PowerAuthEciesDecryptorParameters getEciesDecryptorParameters(String activationId, String applicationKey, String ephemeralPublicKey) throws PowerAuthEncryptionException;
 
     /**
-     * Prepare ECIES data from PowerAuth encryption HTTP header.
+     * Decrypt HTTP request body and construct object with ECIES data. Use the requestType parameter to specify
+     * the type of decrypted object.
      *
-     * @param encryptionHttpHeader PowerAuth encryption HTTP header.
-     * @return PowerAuth ECIES encryption object.
-     * @throws PowerAuthEncryptionException In case PowerAuth encryption HTTP header is invalid.
+     * @param request HTTP request.
+     * @param requestType Request object type.
+     * @return Object with ECIES data.
+     * @throws PowerAuthEncryptionException In case request decryption fails.
      */
-    public PowerAuthEciesEncryption prepareEciesEncryption(String encryptionHttpHeader) throws PowerAuthEncryptionException {
-        // Check for HTTP PowerAuth encryption header
-        if (encryptionHttpHeader == null || encryptionHttpHeader.equals("undefined")) {
-            throw new PowerAuthEncryptionException("POWER_AUTH_ENCRYPTION_INVALID_EMPTY");
+    public <T> PowerAuthEciesEncryption<T> decryptRequest(HttpServletRequest request, Class<T> requestType) throws PowerAuthEncryptionException {
+        // Only POST HTTP method is supported for ECIES
+        if (!"POST".equals(request.getMethod())) {
+            throw new PowerAuthEncryptionException("Invalid HTTP request");
         }
+        // Construct ECIES encryption object from HTTP header
+        final PowerAuthEciesEncryption<T> eciesEncryption = new PowerAuthEciesEncryption<>(request.getHeader(PowerAuthEncryptionHttpHeader.HEADER_NAME));
```

Comment:
We probably need some better handling here, to account for signed encrypted requests.

Reply:
Correct, however I suggest to implement support for signed and encrypted requests as part of another issue. We already have https://github.com/wultra/powerauth-restful-integration/issues/84, let's update it with current requirements.